internal/host/transition.go

```diff
@@ -30,6 +30,39 @@ type transitionHandler struct {
 	eventsHandler eventsapi.Handler
 }
 
+//go:generate mockgen -source=transition.go -package=host -destination=mock_transition.go
+type TransitionHandler interface {
```

Comment:
Made transitionHandler struct conform to an interface to it can be mocked. This mocking allows us to easily test the state machine in isolation